catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/IngestCommand.java

```diff
@@ -452,60 +484,158 @@ private void moveToFailedIngestDirectory(File source) {
+                                        INGEST_LOGGER.error(
+                                                "Thread interrupted while waiting to 'put' metacard: {}",
+                                                e);
+                                    }
                                 }
                             }
                         }
-                    }
-                });
+                    });
+        }
         doneBuildingQueue.set(true);
     }
 
+    private void submitToStorageProvider(List<Metacard> metacardList) {
+        StorageProvider storageProvider = getAllServices(StorageProvider.class).get(0);
+
+        metacardList.forEach((metacard) -> {
+
+            List<File> fileList = metacardFileMapping.get(metacard.getId());
+
+            if (fileList != null) {
+                List<ContentItem> contentItemList = new ArrayList<>();
+                ContentItem contentItem;
+
+                for (File file : fileList) {
+                    ByteSource byteSource = com.google.common.io.Files.asByteSource(file);
+                    String fileName = file.getName()
+                            .split("-")[1];
+
+                    String fragment = null;
+                    if (!file.getPath()
+                            .contains(CONTENT + File.separator + metacard.getId())) {
+                        fragment = StringUtils.substringBetween(file.getPath(),
+                                CONTENT + File.separator,
+                                File.separator + metacard.getId());
+                    }
+                    contentItem = new ContentItemImpl(metacard.getId(),
+                            fragment,
+                            byteSource,
+                            metacard.getContentTypeName(),
+                            fileName,
+                            file.length(),
+                            metacard);
+                    contentItemList.add(contentItem);
+                }
+
+                CreateStorageRequest createStorageRequest = new CreateStorageRequestImpl(
+                        contentItemList,
+                        metacard.getId(),
+                        new HashMap<>());
+                try {
+                    storageProvider.create(createStorageRequest);
+                } catch (StorageException e) {
+                    LOGGER.debug("Unable to create content for {}",
+                            createStorageRequest.getId(),
+                            e);
+                    try {
+                        storageProvider.rollback(createStorageRequest);
+                    } catch (StorageException e1) {
+                        LOGGER.debug("Unable to perform rollback on temporary content for {} ",
+                                createStorageRequest.getId(),
+                                e1);
+                    }
+                } finally {
+                    try {
```

Comment:
You could potentially be calling commit() after rollback() here.  I think this block could be much simpler: https://gist.github.com/dcruver/0e306f49c4e3fb1456611b9c60cb4a87